src/codeflare_sdk/utils/generate_yaml.py

```diff
@@ -141,6 +141,14 @@ def update_image(spec, image):
         container["image"] = image
 
 
+def update_image_pull_secrets(spec, image_pull_secrets):
+    if image_pull_secrets:
+        if "imagePullSecrets" not in spec:
+            spec["imagePullSecrets"] = []
+        for image_pull_secret in image_pull_secrets:
+            spec["imagePullSecrets"].append({"name": image_pull_secret})
```

Comment:
```suggestion
    template_secrets = spec.get("imagePullSecrets", [])
    spec["imagePullSecrets"] = template_secrets + [{"name": x} for x in image_pull_secrets] 
```

Reply:
Sweet, thats a clean alteration. Thanks for the input @KPostOffice